### PR TITLE
Add the get tx status functionality

### PR DESCRIPTION
**Pull request by noambergIL**
update the spec and config
add the get status function
add and order tests
update the http server

### CODE REVIEW

File: bootstrap/httpserver/server_handlers_test.go [hunk 1]
```diff
@@ -107,3 +107,45 @@ func TestHttpServerCallMethod_Error(t *testing.T) {
...
+
+	s := NewHttpServer("", logger, papiMock)
+
+	request := (&client.GetTransactionStatusRequestBuilder{}).Build()
+
+	req, _ := http.NewRequest("POST", "", bytes.NewReader(request.Raw()))
+	rec := httptest.NewRecorder()
+	s.(*server).getTransactionStatusHandler(rec, req)
+
+	require.Equal(t, http.StatusInternalServerError, rec.Code, "should fail with 500")
```
Comment (electricmonk) [inline bootstrap/httpserver/server_handlers_test.go]: did you look at `require.HttpError`?

File: services/publicapi/service.go [hunk 2]
```diff
@@ -145,6 +156,110 @@ func toTxOutput(transactionOutput *txResponse) *services.SendTransactionOutput {
 	return &services.SendTransactionOutput{ClientResponse: response.Build()}
 }
 
+func (s *service) CallMethod(input *services.CallMethodInput) (*services.CallMethodOutput, error) {
```
Comment (electricmonk) [inline services/publicapi/service.go]: consider splitting separate methods to separate files

File: services/publicapi/test/get_transaction_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+func TestGetTransactionStatus_GetPendingStatusFromTxPool(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		harness := newPublicApiHarness(ctx, 1*time.Second)
+
+		txb := builders.Transaction().Builder()
+		txHash := digest.CalcTxHash(txb.Build().Transaction())
+
+		harness.OnGetTransactionSetPending()
```
Comment (electricmonk) [inline services/publicapi/test/get_transaction_test.go]: these methods don't need to be exported
also, maybe name the setup method according to the scenarios they describe:
```
harness.transactionIsInPool(txHash)
harness.transactionNotInPool(txHash)
```

File: bootstrap/httpserver/server_handlers_test.go (hunk 1 above)
Comment (noambergIL) [reply bootstrap/httpserver/server_handlers_test.go]: i looked ... it doesn't seem to have somehting that is binary post in the body.

File: services/publicapi/service.go (hunk 2 above)
Comment (noambergIL) [reply services/publicapi/service.go]: will consider when i do call method

**Pull request merged**